src/hb-ot-layout.cc

```diff
@@ -1703,6 +1703,63 @@ hb_ot_layout_get_size_params (hb_face_t       *face,
 
   return false;
 }
+
+struct hb_position_single_dispatch_t :
+       hb_dispatch_context_t<hb_position_single_dispatch_t, bool>
+{
+  static return_t default_return_value () { return false; }
+  bool stop_sublookup_iteration (return_t r) const { return r; }
+
+  private:
+  template <typename T, typename ...Ts> auto
+  _dispatch (const T &obj, hb_priority<1>, Ts&&... ds) HB_AUTO_RETURN
+  ( obj.position_single (std::forward<Ts> (ds)...) )
+  template <typename T, typename ...Ts> auto
+  _dispatch (const T &obj, hb_priority<0>, Ts&&... ds) HB_AUTO_RETURN
+  ( default_return_value () )
+  public:
+  template <typename T, typename ...Ts> auto
+  dispatch (const T &obj, Ts&&... ds) HB_AUTO_RETURN
+  ( _dispatch (obj, hb_prioritize, std::forward<Ts> (ds)...) )
+};
+
+hb_position_t
+hb_ot_layout_get_optical_bound (hb_font_t      *font,
+				unsigned        lookup_index,
+				hb_direction_t  direction,
+				hb_codepoint_t  glyph)
+{
+  /* This is ugly... */
+  hb_buffer_t buffer;
+  buffer.props.direction = direction;
+  OT::hb_ot_apply_context_t c_apply (1, font, &buffer);
+  const OT::PosLookup &lookup = font->face->table.GPOS->table->get_lookup (lookup_index);
+  hb_glyph_position_t pos = {0};
+  hb_position_single_dispatch_t c;
+  lookup.dispatch (&c, &c_apply, glyph, pos);
+  hb_position_t ret = 0;
+  switch (direction)
```

Comment:
Do you think this is a misuse of direction? What we really want here is a side (ie left/right/top/bottom), not a direction. 
The right bounds will be used in ltr context too...

Reply:
I know... But another enum with exactly four values sounds a bit overkill...

Reply:
Think of it as "left-to-right edge of the glyph".

Reply:
but then, there are no features for top/bottom bounds anyway

Reply:
In theory there could; just needs a feature tag (which could be private) to be named.